internal/gatewayapi/translator.go

```diff
@@ -112,7 +112,7 @@ func (t *Translator) GetRelevantGateways(gateways []*v1beta1.Gateway) []*Gateway
 	var relevant []*GatewayContext
 
 	for _, gateway := range gateways {
-		if gateway.Spec.GatewayClassName == t.GatewayClassName {
+		if gateway != nil && gateway.Spec.GatewayClassName == t.GatewayClassName {
```

Comment:
imho this should ideally not be happening, but rather than skip it, we should either
* log it or
* return early and empty
to surface the underlying issue faster

Reply:
I agree we shouldn't be seeing `nil` gateways, so I'd actually be in favor of doing
```go
if gateway == nil {
  // Log some message here
  panic
}
```